src/cpp/rtps/writer/StatefulWriter.cpp

```diff
@@ -598,13 +623,19 @@ void StatefulWriter::send_changes_separatedly(
 
     for (ReaderProxy* remoteReader : matched_readers_)
     {
+        // If there are no changes for this reader, simply jump to the next one
+        if (!remoteReader->has_changes())
+        {
+            continue;
+        }
+
         if (remoteReader->is_local_reader())
         {
             SequenceNumber_t max_ack_seq = SequenceNumber_t::unknown();
             auto unsent_change_process =
                     [&](const SequenceNumber_t& seqNum, const ChangeForReader_t* unsentChange)
                     {
-                        if (unsentChange != nullptr && unsentChange->isRelevant() && unsentChange->isValid())
+                        if (unsentChange != nullptr && unsentChange->isValid())
```

Comment:
As per our conversation, do not remove the `isRelevant()` check.

Also applicable to the rest of the file where the check has been removed.

Reply:
It turns out that only relevant `CacheChange_t` manage to get an associated `ChangeForReader` in the ReaderProxy. Thus, we realized all isRelevant() calls will always return true and are useless in `send_changes_separatedly` context.